osu.Server.Spectator/Hubs/MultiplayerQueue.cs

```diff
@@ -263,28 +269,34 @@ private async Task updatePlaylistOrder(IDatabaseAccess db)
                     break;
 
                 case QueueMode.AllPlayersRoundRobin:
-                    var itemsByPriority = new List<(MultiplayerPlaylistItem item, int priority)>();
+                    orderedActiveItems = new List<MultiplayerPlaylistItem>();
+
+                    bool isFirstSet = true;
+                    var firstSetOrderByUserId = new Dictionary<int, int>();
 
-                    // Assign a priority for items from each user, starting from 0 and increasing in order which the user added the items.
-                    foreach (var group in room.Playlist.Where(item => !item.Expired).OrderBy(item => item.ID).GroupBy(item => item.OwnerID))
+                    // Group each user's items in order of addition.
+                    var userItemGroups = room.Playlist.Where(item => !item.Expired).OrderBy(item => item.ID).GroupBy(item => item.OwnerID);
+
+                    foreach (IEnumerable<MultiplayerPlaylistItem> set in interleaveMany(userItemGroups))
```

Comment:
Would rename "set" to "item". "set" to me reads like it's a "set of items" which makes it hard to mentally parse this function.

Reply:
The `set` here *is* an enumerable here, though, not a single item. In the first iteration `set` is a set composed of each individual player's oldest queued playlist items, in the second - the second oldest queued items, in the third - third oldest, etc.

Reply:
This is correct. It's also why I explicitly specified the type.

Reply:
Maybe separating it to a local may enhance the readability a bit? (not sure on the given name though)
```csharp
var groupsPerPriority = Interleave(userItemGroups);

foreach (var itemGroup in groupsPerPriority)
{

}
```

I was also confused into thinking that the foreach loop was iterating per item rater than per set mainly due to the method name including `Many` which makes me think it returns a flattened `IEnumerable<T>` type.

Reply:
I think that would confuse me more...